### DISCUSSION TITLE
Orchard core  Users, user groups, roles, site users, permissions

### DISCUSSION THREAD

**Discussion by pgy866** [OrchardCMS/OrchardCore#15968] [Q&A]
Users, user groups, roles, site users, permissions
Don't see the user group? How do you do that?
Global database? Site database?

**Comment by CSurieux** [comment]
It seems related to ##5529

**Comment by pgy866** [comment]
yes,It's related. What if you add a global database and a site database? This also enables tenant nesting for unlimited levels of tenants

**Comment by CSurieux** [comment]
Sorry I am not sure to understand what you ask ?
My suggestion was just to ask for an interface to manage ownership, allowing more extensions later as teams.

**Comment by sebastienros** [comment]
Can someone find the issue where we talked about the notion of Teams or Groups, different than roles?

**Comment by JoshLefebvre** [comment]
The site I am currently working on will require user groups and will likely be working on a custom module to accomplish this. Seeing as this has been added to the backlog I would be happy to try and tackle this if the functionality aligns. 

My use case would be to have a series of courses (content items) and then give a group of users (a classroom of students) access to this course. So different then permissions, in that I don't want to create a new permission for every new course as this will be something dynamic. 

**My Approach**
One approach would be to follow something similar to the Menu Content Item.
ContentType Name: "UserGroup"
1, TitlePart: Name of the group
2. UserListPart: List\<Users\> Users
3, UserGroupPart: Meta fields related to user groups. e.g. description

**Comment by JoshLefebvre** [comment]
Unlike menus this should likely be placed under the security tab.

**Comment by JoshLefebvre** [comment]
I've gone ahead and created my own OC User Groups module that mostly fits my use case. It is quite primitive but I will not able to wait for it to be implemented in the road map as I need this functionality for a feature I am working on. If anyone is for a simple module to create groups feel free to take a look at my repo:

https://github.com/JoshLefebvre/OrchardCore.UserGroups

**Reply by hishamco** [reply]
@JoshLefebvre It's too late but I came accross such situation in Orchard Core Contrib (OCC) https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/issues/148 AFAIK what you did is allowing a group of users to be attached into your content items, but you will find yourself repeating this if you go for another content item

What I come up with is a notion of `UserGroup` which is organizing the users into  one or more groups, something similar to roles. Hope to publish the package ASAP and the could will be a part of OCC

Another things this module could extends the  `OCC.ContentPermissions` where group(s) could be selected to protect the content item
